web/assets/js/repository.js

```diff
@@ -0,0 +1,22 @@
+$(document).ready(function() {
+    $('#repository-star').click(function() {
+        var link = $(this);
+        $.ajax(
+            $(this).attr('href'),
+            {
+                method: 'PUT',
+                dataType: 'json',
+                success: function(data) {
+                    var icon = link.find('.glyphicon');
+                    if (data.starred) {
+                        icon.addClass('glyphicon-star').removeClass('glyphicon-star-empty');
+                    } else {
+                        icon.addClass('glyphicon-star-empty').removeClass('glyphicon-star');
+                    }
+                }
+            }
+        );
+
+        return false;
```

Comment:
use `event.preventDefault();` instead of `return false;`

do not forget to add `event` as input parameter of the function